### DISCUSSION TITLE
Add TTS (text to speech) back to lovelace media player

### DISCUSSION THREAD

**Discussion by dredjohn21** [home-assistant/frontend#11923] [Dashboard]
![870697B3-2D30-43EC-B0B6-32D8973D4E12](https://user-images.githubusercontent.com/95455576/156683433-297d4b44-c0b3-42fc-a859-c795c68b7ce2.jpeg)

The latest update removed the ability to quickly enter text and broadcast to a speaker.  We used this frequently to notify family members in the basement.  Can we please bring back?  There is no easy alternative.  Need a quick solution from lovelace overview page.

**Comment by mikje** [comment]
I downgraded to 2022.2.9 where it is working
use the command in putty:
ha core update --version 2022.2.9

**Comment by cooljimy84** [comment]
Please, I understand having it in the media part, but removing it makes it more clicks to achieve what we could do quickly before.

**Comment by tyjtyj** [comment]
Agree, this is such useful function.. 

i dont even have media browser enabled

**Comment by vazqued2** [comment]
It also broke the TTS functionality for my google home speakers for some reason.

**Comment by drthanwho** [comment]
This could easily by done alternatively by using an input_text helper and passing it's contents via an automation/script to the call service. Then you can have an input in your lovelace to type the text in directly. If you have multiple speakers can even additionally add input_select to choose between all your speakers.

This is for mobile_app TTS but you get the point
![image](https://user-images.githubusercontent.com/22405505/159693396-2a71b975-d0b4-48f3-84a3-127492853f9c.png)

**Comment by HVR88** [comment]
"could easily be done" - I'd love to see some people in the real-world working a real job.  hahaha.

Removing the TTS option from individual cards only to replace it with a message saying TTS isn't available there is counter-productive. If you're going to have that message you may as well have a link to the Media Browser. And if you go that far, you might as well go back to square one and put the original control back.

Look, TTS is not a "Media Browsing" function at all. TTS doesn't belong in the Media Browser UI. Its implementation in that spot is awkward, non-intuitive and at the moment buggy that it's missing any number of player entities, leaving people without the ability to control many of their devices (see https://github.com/home-assistant/core/issues/67518) .

Compromise. Go all-in on the Media Browser, but leave the TTS control on individual cards for quick one-off access. Everyone knows that (once the bugs are fixed) automations of all sorts can be created to do so much more, but none of it is a replacement for quick-access built into the original entity card.

**Comment by loadamasta** [comment]
I have the same issue. No chance to get TTS working on my players via the media section. This "update" is one of the worst "improvment" I've ever seen in home assistant. It's simply a very huge step back. Want not to be rude but this "improvment" is bullsh*#*! Please, bring back the functionality as before, thank you.

**Comment by MrThiemann** [comment]
I faced the same problem today.
"Unfortunately" I only have Alexa Echo's. Which makes the whole thing a bit "difficult" anyway with the media and your own MP3s. ;)

However, I did find one:
you can record your TTS directly in the card.
It doesn't always look pretty, but it serves its purpose!

<img width="777" alt="Bildschirmfoto 2022-04-09 um 18 01 37" src="https://user-images.githubusercontent.com/54147030/162581880-4f4bd52f-0b47-4d81-ac8d-deb993ed91e0.png">

Alternatively, you could of course also build a media_card with a custom:button-card and then go into more detail regarding TTS in a "tap_action".

**Reply by Budilnik** [reply]
Thanks for the tip! Perhaps you now how to send a command to the speaker, not an announcement?

**Reply by MrThiemann** [reply]
what do you mean by "command"?

**Reply by Budilnik** [reply]
I use a different smart speaker (not Alexa), but as far as I know they can all receive smart home commands (for example, "Turn on the light in the hall")

**Reply by MrThiemann** [reply]
that's fine. And I claim that everyone can do that. And definitely with NodeRed.
I'm currently doing some magic in NodeRed. There are already some ready-made ones - but either they don't work properly for me, or I'm missing something. So now you have to have something of your own =) which speaker do you use?

**Reply by Budilnik** [reply]
I have Yandex.Station and I found a solution in the description of its native integration. The command can be sent through the HA script.

**Comment by Kalembang** [comment]
Why would this be removed!!! The new process is soooooo annoying!

**Comment by ibizaman** [comment]
I agree it's not easy, but this is a summary of what you would need to see a text input in the dashboard and have some speaker say it out loud when the text changes (when you press enter).

I'm using home assistant 2023.6.0

1. Settings > Devices & Services > Helpers (tab) > Create Helper > Text
Name: Speak Text
Icon: mdi:comment-text-outline

This creates a new Entity with ID `input_text.speak_text`.

2. Settings > Automations & Scenes > Create Automation > Create new automation (empty)
    1. Add trigger > State > Set Entity to `input_text.speak_text` and leave optional fields (attributes, etc.) empty.
    2. Add action > Call service > Edit in YAML (because we can't use templates in the Visual Editor) > Paste:
        ```yaml
        service: tts.speak
        data:
          cache: true
          media_player_entity_id: media_player.my_speaker
          message: "{{ states('input_text.speak_text') }}"
        target:
          entity_id: tts.piper
        ```
        Replace `media_player.my_speaker` by your media player entity ID.

4. In your dashboard, Edit > Add Card > By Entity (tab) > Search for `input_text.speak_text` > Click checkbox next to it > Continue > Add to dashboard.

Now, whenever you go to your dashboard and you update the text and press enter, you will hear it.
